pages/docs/reference/middleware/lifecycle.mdx

```diff
@@ -182,6 +182,17 @@ The `init()` function can return functions for two separate lifecycles to hook i
                          </Property>
                     </Properties>
                </Property>
+               <Property name="finished" type="function" version="v3.21.0+">
+                    Called when the function has finished executing and has returned a final response that will end the run which could be either a successful or erroring response.
+
+                    This is not guaranteed to be called on every execution, and may be called multiple times if many parallel executions reach the end of the function or during retries.
```

Comment:
wdyt

```suggestion
This function is not guaranteed to be called on every execution. It may be called multiple times if there are many parallel executions or during retries.
```